airbyte-integrations/connectors/source-salesforce/unit_tests/api_test.py

```diff
@@ -80,30 +83,26 @@ def test_stream_contains_unsupported_properties_by_bulk(stream_config, stream_ap
     assert not isinstance(stream, BulkSalesforceStream)
 
 
-@pytest.mark.parametrize("item_number", [0, 15, 2000, 2324, 3000])
-def test_bulk_sync_pagination(item_number, stream_config, stream_api):
+def test_bulk_sync_pagination(stream_config, stream_api, requests_mock):
```

Comment:
could you add a regression test for the bug we ran into where if we ask for records between oct1-oct10 but receive a record whose systemModStamp is `today` then the state should not advance beyond october 10?

Reply:
sorry could you clarify where that test is? I'm not seeing a test with incremental sync mode where the input state provides a cursor value with time = `today`, I see one for full refresh sync mode, is that the one?

Reply:
I've updated the test `test_bulk_stream_request_params_states`:
it works in incremental mode now and covers the OC case we met: record with cursor=datetime.now inside some slice.
Test asserts both request params in all slices and produced states